src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs

```diff
@@ -12,204 +12,191 @@
-        protected readonly ILoggingAdapter _logger;
-        private Flow<JournalRow, Util.Try<(IPersistentRepresentation, IImmutableSet<string>, long)>, NotUsed> deserializeFlow;
-        private Flow<JournalRow, Util.Try<ReplayCompletion>, NotUsed> deserializeFlowMapped;
-
-        protected BaseByteArrayJournalDao(IAdvancedScheduler sched,
-            IMaterializer materializerr,
+        protected BaseByteArrayJournalDao(
+            IAdvancedScheduler scheduler,
+            IMaterializer materializer,
             AkkaPersistenceDataConnectionFactory connectionFactory,
-            JournalConfig config, ByteArrayJournalSerializer serializer, ILoggingAdapter logger) : base(
-            sched, materializerr, connectionFactory)
+            JournalConfig config, 
+            ByteArrayJournalSerializer serializer, 
+            ILoggingAdapter logger) 
+            : base(scheduler, materializer, connectionFactory)
         {
-            _logger = logger;
-            _journalConfig = config;
-            logicalDelete = _journalConfig.DaoConfig.LogicalDelete;
+            Logger = logger;
+            JournalConfig = config;
+            LogicalDelete = JournalConfig.DaoConfig.LogicalDelete;
             Serializer = serializer;
-            deserializeFlow = Serializer.DeserializeFlow();
-            deserializeFlowMapped = Serializer.DeserializeFlow().Select(MessageWithBatchMapper());
+            _deserializeFlow = Serializer.DeserializeFlow();
+            _deserializeFlowMapped = Serializer.DeserializeFlow().Select(MessageWithBatchMapper());
+            
             //Due to C# rules we have to initialize WriteQueue here
             //Keeping it here vs init function prevents accidental moving of init
             //to where variables aren't set yet.
             WriteQueue = Source
-                .Queue<WriteQueueEntry
-                >(_journalConfig.DaoConfig.BufferSize,
-                    OverflowStrategy.DropNew)
-                .BatchWeighted(_journalConfig.DaoConfig.BatchSize,
-                    cf=>cf.Rows.Count,
-                    r => new WriteQueueSet(
-                        new List<TaskCompletionSource<NotUsed>>(new[]
-                            {r.TCS}), r.Rows),
-                    (oldRows, newRows) =>
-                    {
-                        oldRows.TCS.Add(newRows.TCS);
-                        oldRows.Rows = oldRows.Rows.Concat(newRows.Rows);
-                        return oldRows; //.Concat(newRows.Item2).ToList());
-                    })
-                .SelectAsync(_journalConfig.DaoConfig.Parallelism,
-                    async (promisesAndRows) =>
+                .Queue<WriteQueueEntry>(JournalConfig.DaoConfig.BufferSize, OverflowStrategy.DropNew)
+                .BatchWeighted(
+                    JournalConfig.DaoConfig.BatchSize,
+                    cf => cf.Rows.Count,
+                    r => new WriteQueueSet(ImmutableList.Create(new[] {r.Tcs}), r.Rows),
```

Comment:
Per my other comment I wonder if we benchmarked this change against both basic and heavy parallelism cases?